crates/abq_utils/src/net_protocol.rs

```diff
@@ -307,6 +307,31 @@ pub mod workers {
         }
     }
 
+    /// ABQ-internal-ID for a grouping
+    /// In order to do file-based allocation to workers, we need to have a way of
+    /// knowing which tests are in which file. We use this group id as a proxy for that.
+    /// Eventually, these groupings will be assigned to specific workers
+    #[derive(Serialize, Deserialize, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
+    pub struct GroupId(pub [u8; 16]);
```

Comment:
We don't need to do this now, but given that this never travels over the network (and neither does the flat manifest), I wonder if we should structure this a different way:

- remove `flatten` as a method on `Manifest`, and instead add a module in `abq_queue` that provides that function.
- localize `GroupId` in that module so we don't need to create it in the common abq_utils.